### PR TITLE
make sure only highlighted results are shown.

### PR DESCRIPTION
**Pull request by evans-g-crsj**

### CODE REVIEW

File: modules/components/src/AdvancedFacetView/utils.js [hunk 1]
```diff
@@ -97,11 +97,19 @@ const filterDisplayTreeDataBySearchTerm = ({
 }) => {
   const shouldBeIncluded = ({ title, path, children }) => {
     const inTitle = title.match(strToReg(searchTerm));
+    if (inTitle) {
```
Comment (evans-g-crsj) [inline modules/components/src/AdvancedFacetView/utils.js:100]: Less compact but it avoids computing expensive stuff unnecessarily and it's not really less readable in my option.

File: modules/components/src/Aggs/TermAgg.js [hunk 2]
```diff
@@ -76,13 +76,14 @@ const MoreOrLessButton = ({ howManyMore, isMore, onClick }) => (
 );
 
 const decorateBuckets = ({ buckets, searchText }) => {
+  const matcher = strToReg(searchText);
   const namedFilteredBuckets = buckets
-    .map(b => ({ ...b, name: b.key_as_string || b.key }))
```
Comment (evans-g-crsj) [inline modules/components/src/Aggs/TermAgg.js:80]: I switch the order from map -> filter TO filter -> map

File: modules/components/src/AdvancedFacetView/index.js [hunk 3]
```diff
@@ -130,7 +131,7 @@ export default class AdvancedFacetView extends React.Component {
           aggregations,
         }).displayTreeDataWithValue
       : displayTreeData;
-
+      const filteredAggregations = matchingAggregations(aggregations, searchTerm);
```
Comment (evans-g-crsj) [inline modules/components/src/AdvancedFacetView/index.js:134]: It's a patch, it doesn't solve the root problem, but it's the safest and simplest solution I could think of. I've been told that this whole part could be refactored so I didn't want to invest too much in it.

File: modules/components/package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@kfarranger/components",
-  "version": "1.3.3",
+  "version": "^1.4.0",
```
Comment (jberube) [inline modules/components/package.json]: Just put 1.4.0.

Use ^1.4.0 in the package.json of kf-portal-ui

**Review by jberube — CHANGES_REQUESTED** [review]

**Pull request merged**